plugins/performance-lab/includes/admin/plugins.php

```diff
@@ -86,10 +88,9 @@ function perflab_query_plugin_info( string $plugin_slug ) {
 		}
 
 		if ( ! isset( $all_performance_plugins[ $current_plugin_slug ] ) ) {
-			return new WP_Error(
-				'plugin_not_found',
-				__( 'Plugin not found in WordPress.org API response.', 'performance-lab' )
-			);
+			// Cache the fact that the plugin was not found.
+			$plugins[ $current_plugin_slug ] = false;
```

Comment:
Instead of storing `false` here, what about storing the `WP_Error` instance? As it stands right now, there are two conditions resulting in `false` being logged, which conflates these two error scenarios which we tried to get rid of in #1651 to assist with debugging.

Reply:
Overall this makes sense to me. We probably shouldn't store the `WP_Error` _instance_ though, but just its data, since storing PHP class instances is a bit unpredictable with what would happen in different object cache implementations.